### ISSUE TITLE
SkyDNS silently failing after arbitrary amount of time

### ISSUE THREAD

**Issue by bloo** [kubernetes/kubernetes#23601]
This leaves any new services we deploy unregistered with the cluster DNS system, and our ingress (nginx) fails when looking for those upstream services. The kube2sky log starts going crazy with:

```
E0327 07:48:00.194028       1 reflector.go:136] Failed to list *api.Endpoints: the server has asked for the client to provide credentials (get endpoints)
E0327 07:48:01.199254       1 reflector.go:136] Failed to list *api.Service: the server has asked for the client to provide credentials (get services)
E0327 07:48:01.200430       1 reflector.go:136] Failed to list *api.Endpoints: the server has asked for the client to provide credentials (get endpoints)
E0327 07:48:02.203628       1 reflector.go:136] Failed to list *api.Service: the server has asked for the client to provide credentials (get services)
E0327 07:48:02.203814       1 reflector.go:136] Failed to list *api.Endpoints: the server has asked for the client to provide credentials (get endpoints)
E0327 07:48:03.207383       1 reflector.go:136] Failed to list *api.Service: the server has asked for the client to provide credentials (get services)
E0^Che client to provide credentials (get endpoints)
E0327 07:48:07.218230       1 reflector.go:136] Failed to list *api.Service: the server has asked for the client to provide credentials (get services)
E0327 07:48:07.223236       1 reflector.go:136] Failed to list *api.Endpoints: the server has asked for the client to provide credentials (get endpoints)
```

**Comment by a-robinson** [comment]
Can you give more detail on what guide you used to deploy your cluster?

@kubernetes/goog-cluster

**Comment by bprashanth** [comment]
You absolutely need to be able to talk to the apiserver for kube-dns to work. It looks like you're not able to. Did it ever work?

**Comment by karthikus** [comment]
I am facing the same issue the api server works with master/workers and I also could deploy pods. The only thing that does not work is the proxy and I also get the exact same error. What puzzles me is that I do not provide any cert/keys location in the dns pod file ([Doc link](https://coreos.com/kubernetes/docs/latest/deploy-addons.html). Any suggestions ?

**Comment by bprashanth** [comment]
All pods in the default and kube-system namespaces have the capacity to talk to the master using a token mounted at /var/run/secrets/kubernetes.io/serviceaccount by a service account. Make sure you bootstrapped your cluster with the right certs (http://kubernetes.io/docs/getting-started-guides/scratch/#preparing-certs) and if you're deploying HA, that all the master components have the same private key

**Comment by karthikus** [comment]
@bprashanth Thanks for your response. Yes, in my case, I updated the keys and services multiple times and that might be the issue with the old token left behind. 
1. When we update the keys or do a redeployment of kubernetes, should we remove that directory ?
I tried a clean deployment and I could not see the errors anymore but the following dns test did not work:

```
kubectl exec busybox -- nslookup kubernetes.default 

Server:    12.1.2.10
Address 1: 12.1.2.10

nslookup: can't resolve 'kubernetes.default'
error: error executing remote command: Error executing command in container: Error executing in Docker Container: 1
```
The API server and the DNS seems to have no error in the logs.
